Looks like the identity change has some trouble in this branch, /cc @gandro 
```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x171e4a2]

goroutine 13 [running]:

github.com/cilium/cilium/pkg/identity/cache.(*localIdentityCache).lookupByID(0x0, 0xc0000003e8, 0x0)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/local.go:170 +0x42

github.com/cilium/cilium/pkg/identity/cache.(*CachingIdentityAllocator).LookupIdentityByID(0xc0001271a0, 0x2057760, 0xc000120018, 0x3e8, 0x10)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/cache.go:245 +0x90

github.com/cilium/cilium/pkg/proxy.(*defaultEndpointInfoRegistry).FillEndpointIdentityByID(0x306fba0, 0x3e8, 0xc00002ec90, 0xc0002549e0)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/epinfo.go:50 +0x57

github.com/cilium/cilium/pkg/proxy/logger.(*LogRecord).fillIngressSourceInfo(0xc00002ec40, 0xc00002ec90, 0xc000624db0, 0x3e8)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:157 +0x74

github.com/cilium/cilium/pkg/proxy/logger.logTags.Addressing.func1(0xc00002ec40)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:234 +0x2d3

github.com/cilium/cilium/pkg/proxy/logger.(*LogRecord).ApplyTags(0xc00002ec40, 0xc000624eb0, 0x1, 0x1)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/logger/logger.go:286 +0x49

github.com/cilium/cilium/pkg/proxy.(*kafkaRedirect).handleRequest(0xc000359170, 0xc0002f2090, 0xc0002f24e0, 0xc0002be0c0, 0x2022000, 0xc0002f2060, 0x3e8, 0xc0002544b0, 0xf)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:316 +0x3dd

github.com/cilium/cilium/pkg/proxy.(*kafkaRedirect).handleRequests(0xc000359170, 0xc000067500, 0xc0002f2090, 0xc0002be000, 0xc000625e70)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:443 +0x742

github.com/cilium/cilium/pkg/proxy.(*kafkaRedirect).handleRequestConnection(0xc000359170, 0xc0002f2090)

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:498 +0x1ee

created by github.com/cilium/cilium/pkg/proxy.(*kafkaListener).Listen

	/home/travis/gopath/src/github.com/cilium/cilium/pkg/proxy/kafka.go:130 +0x922

FAIL	github.com/cilium/cilium/pkg/proxy	0.046s

FAIL

Makefile:209: recipe for target 'unit-tests' failed

make: *** [unit-tests] Error 1

--- a/pkg/datapath/linux/node.go
+++ b/pkg/datapath/linux/node.go
@@ -700,7 +700,7 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		oldKey = oldNode.EncryptionKey
 	}
 
-	if n.nodeConfig.EnableIPSec && !n.subnetEncryption() {
+	if n.nodeConfig.EnableIPSec && !n.subnetEncryption() && !n.nodeConfig.EncryptNode {
 		n.enableIPsec(newNode)
 		newKey = newNode.EncryptionKey
 	}

